feat: start apple pay doc and new sidebar (#346)
Basis-Theory/developers.basistheory.com: docs/guides/apple-pay/accept.mdx

```shell showLineNumbers
curl "https://api.basistheory.com/reactors" \
  -H "BT-API-KEY: <MANAGEMENT_API_KEY>" \
  -H "Content-Type: application/json" \
  -X "POST" \
  -d '{
    "name": "Apple Pay Reactor",
    "code": '"$(echo $reactor_code | jq -Rsa .)"',
    "configuration": {
      "MERCHANT_IDENTITY_CERTIFICATE_PEM": "-----BEGIN CERTIFICATE-----\nMIIEdTCCBBugAwIBAgIIc5onAJBqqpk...",
      "PAYMENT_PROCESSING_PRIVATE_KEY_PEM": "-----BEGIN EC PARAMETERS-----\nBggqhkjOPQMBBw==..."
    },
    "application": {
      "id": "<PRIVATE_APPLICATION_ID>"
    }
  }'
```

Important things to notice in the request above:

1. `<MANAGEMENT_API_KEY>` is the Management Application Key, used to authenticate the request
2. `code` is passed in plain text form
3. `<PRIVATE_APPLICATION_ID>` is the Private Application Id, used to inject `req.bt` in the Reactor code
4. Merchant Identifier Certificate and Payment Processing Private Key are passed in plain text form as configuration entries

<Alert>Save the <code>id</code> from the created Reactor as it will be used later to invoke it.</Alert>

Done! These are all the resources necessary. Let's see how to use them.

## Decrypting PKPaymentToken

> In this section, we will explore how the integration looks like for both the client and server applications.

Reviewer: ```suggestion
In this section, we will explore the integration for both the client and server applications.
```

### Creating a Session

> Chose your preferred language for your client application, and click below for detailed instructions on how to install and configure the SDK.

Reviewer: ```suggestion
Choose your preferred language for your client application, and click below for detailed instructions on installing and configuring the SDK.
```

To invoke a Reactor, first we need to create a [Session](/docs/api/applications/sessions) to grant temporary access to our Public Application. It is good practice to create the session when the user enters the checkout page, so it is ready by the time they click on the "Pay" button.

<Tabs className="bt-tabs" queryString="frontend">
  <TabItem value="web" label="Web">

```javascript showLineNumbers title=index.js
import { BasisTheory } from '@basis-theory/basis-theory-js';

let bt;
let sessionKey;

// initializes Basis Theory
bt = await new BasisTheory().init("<PUBLIC_API_KEY>");

// highlight-next-line
const session = await bt.sessions.create();
sessionKey = session.sessionKey;
```

  </TabItem>
  <TabItem value="ios" label="iOS">

```swift showLineNumbers title=ViewController.swift
import BasisTheory

private var sessionKey: String = ""

// highlight-next-line
SessionsAPI.create(apiKey: "<PUBLIC_API_KEY>") { data, error in
    self.sessionKey = data!.sessionKey!
    let nonce = data!.nonce!
}
```

  </TabItem>
</Tabs>

<Alert>Be sure to replace <code>&lt;PUBLIC_API_KEY></code> with the Public API Key you created previously.</Alert>

### Authorizing a Session

> Sessions must be [authorized](/docs/api/applications/sessions#authorize-session) by a Private Application to perform any protected actions. In our case, we want to allow it to invoke the Apple Pay Reactor to decrypt and tokenize the card information.

Reviewer: ```suggestion
Sessions must be [authorized](/docs/api/applications/sessions#authorize-session) by a Private Application to perform any protected actions. In our case, we need to allow invoking the Apple Pay Reactor to decrypt and tokenize the card information.
```

  console.log(`Server listening on port ${port}`);
});
```

  </TabItem>
  <TabItem value="python" label="Python">

In this example, we are using [Basis Theory SDK](/docs/sdks/server-side/python) and [Flask Framework](https://flask.palletsprojects.com/).

```python showLineNumbers
import os
from flask import Flask, request

import basistheory
# highlight-start
from basistheory.api import sessions_api
from basistheory.model.access_rule import AccessRule
from basistheory.model.condition import Condition
from basistheory.model.authorize_session_request import AuthorizeSessionRequest
# highlight-end

app = Flask(__name__)

# highlight-start
@app.route('/authorize', methods=['POST'])
def authorize_session():
  body = request.get_json()
  with basistheory.ApiClient(configuration=basistheory.Configuration(api_key="<PRIVATE_API_KEY>")) as api_client:
    session_client = sessions_api.SessionsApi(api_client)

    session_client.authorize(authorize_session_request=AuthorizeSessionRequest(
      nonce = body.get("nonce"),
      permissions = [ "token:use" ]
    ))

  return '', 204
# highlight-end

if __name__ == '__main__':
    app.run(port=4242)
```

  </TabItem>

</Tabs>

> <Alert type={Alerts.WARNING}>On a production environment, your endpoint to authorize sessions should be behind your own authorization scheme.</Alert>

Reviewer: ```suggestion
<Alert type={Alerts.WARNING}>In production environments, your endpoint to authorize sessions should be behind your authorization scheme.</Alert>
```

    let session = URLSession.shared

    var request = URLRequest(url: url)
    request.httpMethod = "POST"

    do {
        request.httpBody = try JSONSerialization.data(withJSONObject: parameters, options: .prettyPrinted)
    } catch let error {
        completion(nil, error)
    }

    request.addValue("application/json", forHTTPHeaderField: "Content-Type")
    request.addValue("application/json", forHTTPHeaderField: "Accept")

    let task = session.dataTask(with: request, completionHandler: { data, response, error in
        guard error == nil else {
            completion(nil, error)
            return
        }

        do {
            let json = try JSONSerialization.jsonObject(with: data!, options: .mutableContainers) as? [String: Any]
            completion(json, nil)
        } catch let error {
            completion(nil, error)
        }
    })

    task.resume()
}
// highlight-end

SessionsAPI.create(apiKey: "<PUBLIC_API_KEY>") { data, error in
    self.sessionKey = data!.sessionKey!
    let nonce = data!.nonce!
    // highlight-start
    self.authorizeSession(nonce: nonce) { result, error in

    }
    // highlight-end
}
```

  </TabItem>
</Tabs>

> <Alert type={Alerts.WARNING}>Sessions have a default timeout of 3 minutes before they can be used. You can customize the `expires_at` value when [authorizing the session](/docs/api/applications/sessions#authorize-session), to grant the user more time to customize options in the Apple Pay Sheet and approve the payment.</Alert>

Reviewer: ```suggestion
<Alert type={Alerts.WARNING}>Sessions have a default timeout of 3 minutes before expiration. You can customize the `expires_at` value when [authorizing the session](/docs/api/applications/sessions#authorize-session), to ensure users have time to customize options in the Apple Pay Sheet and approve the payment.</Alert>
```

### Invoking the Reactor

> Once the user has authorized the payment with Touch ID, Face ID or passcode, the Apple Pay Session will invoke the payment authorization event handler, passing the `PKPaymentToken` that we will post to the Reactor.

Reviewer: ```suggestion
Once the user has authorized the payment with Touch ID, Face ID, or passcode, the Apple Pay Session will invoke the payment authorization event handler, passing the `PKPaymentToken`, which we will post to the Reactor.
```

func paymentAuthorizationController(_ controller: PKPaymentAuthorizationController, didAuthorizePayment payment: PKPayment, handler completion: @escaping (PKPaymentAuthorizationResult) -> Void) {
// highlight-start
    let paymentData = try? JSONSerialization.jsonObject(with: payment.token.paymentData)

    let reactRequest = ReactRequest(args: [
        "applePayToken": [
            "paymentMethod": [
                "displayName": payment.token.paymentMethod.displayName,
                "network": payment.token.paymentMethod.network?.rawValue,
                "type": payment.token.paymentMethod.type.rawValue
            ],
            "transactionIdentifier": payment.token.transactionIdentifier,
            "paymentData": paymentData
        ]
    ])

    ReactorsAPI.reactWithRequestBuilder(id: UUID(uuidString: "<REACTOR_ID>")!, reactRequest: reactRequest).addHeader(name: "BT-API-KEY", value: "<SESSION_KEY>").execute { result in
        do {

        } catch {
            print(error)
        }
    }
// highlight-end
}
```

  </TabItem>
</Tabs>

<Alert>Be sure to replace <code>&lt;REACTOR_ID></code> with the Reactor Id you created previously.</Alert>

## Processing Payment

> Now that the Apple Payment Token has been decrypted and the payment information (DPAN and 3DS cryptogram) have been tokenized, send the tokens to your backend and store them in your database. You will use their identifiers to route the detokenized data to the Payment Processor of your choice.
> 
> Each Payment Processor have their own specification for receiving DPANs and 3DS cryptograms for CITs, or just DPANs for MITs. Here are a few things to know when looking into their API specification:
> 
> - Payment Processor usually have their own Apple Pay solution, which will implement both frontend and backend portions. While it may feel like an easier technical integration, it locks in the merchant and their tokens in their ecosystem.
> - It is not surprising that this sort of payment method acceptance is either buried into Processors technical documentation or not publicly available at all.
> - DPAN acts essentially as Network Tokens downstream in the transaction processing workflow, accompanied by the transaction cryptogram and Electronic Commerce Indicator (ECI).
> 
> Take a look on the [Process Card Payments](/docs/guides/share/process-card-payments#send-the-data) guide to learn how to forward tokens to Payment Processors using the Basis Theory Proxy. If you have a hard time figuring out when and which Processor's API endpoints to call, don't hesitate to [contact us](https://basistheory.com/contact). We are happy to support you through your payments journey.
> 
> [Apple Pay requires merchants to complete the payment in 30 seconds](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778020-onpaymentauthorized#discussion), however in many cases that is not possible in practice. So once you have tokenized the payment information, you can call the completion handler to give the user the green check but keep processing in the background.

Reviewer: ```suggestion
Now that the Apple Payment Token has been decrypted and the payment information (DPAN and 3DS cryptogram) have been tokenized, send the tokens to your backend and store them in your database. You will use their identifiers to route the detokenized data to the Payment Processor of your choice.

Each Payment Processor has their specification for receiving DPANs and 3DS cryptograms for CITs, or just DPANs for MITs. Here are a few things to know when looking into their API specification:

- Payment Processor usually have their own Apple Pay solution, which will implement both frontend and backend portions. While it may feel like an easier technical integration, it locks in the merchant and their tokens in their ecosystem.
- It is not surprising that this sort of payment method acceptance is either buried in the Processor's technical documentation or private.
- DPAN acts as a Network Token downstream in the transaction processing workflow, accompanied by the transaction cryptogram and Electronic Commerce Indicator (ECI).

Look at the [Process Card Payments](/docs/guides/share/process-card-payments#send-the-data) guide to learn how to forward tokens to Payment Processors using the Basis Theory Proxy. If you have difficulty figuring out when and which Processor's API endpoints to call, feel free to [contact us](https://basistheory.com/contact). We are happy to support you through your payment journey.

[Apple Pay requires merchants to complete the payment in 30 seconds](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778020-onpaymentauthorized#discussion); however, in many cases, that is not possible in practice. So once you have tokenized the payment information, you can call the completion handler to give the user the green check but keep processing in the background.
```
